scrypto-compiler/src/lib.rs

```diff
@@ -0,0 +1,910 @@
+        match self {
+            Profile::Release => vec![String::from("--release")],
+            Profile::Debug => vec![],
+        }
+    }
+    fn as_directory_name(&self) -> String {
+        match self {
+            Profile::Release => String::from("release"),
+            Profile::Debug => String::from("debug"),
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct ScryptoCompilerBuilder {
+    input_params: ScryptoCompilerInputParams,
+}
+
+impl ScryptoCompilerBuilder {
+    pub fn profile(&mut self, profile: Profile) -> &mut Self {
+        self.input_params.profile = profile;
+        self
+    }
+
+    pub fn env(&mut self, name: &str, value: &str) -> &mut Self {
+        self.input_params
+            .set_environment_variables
+            .push((name.to_string(), value.to_string()));
+        self
+    }
+
+    pub fn unset_env(&mut self, name: &str) -> &mut Self {
+        self.input_params
+            .unset_environment_variables
+            .push(name.to_string());
+        self
+    }
+
+    pub fn feature(&mut self, name: &str) -> &mut Self {
+        self.input_params.features.push(name.to_string());
+        self
+    }
+
+    pub fn package(&mut self, name: &str) -> &mut Self {
+        self.input_params.package = Some(name.to_string());
+        self
+    }
+
+    pub fn target_directory(&mut self, directory: impl Into<PathBuf>) -> &mut Self {
+        self.input_params.target_directory = Some(directory.into());
+
+        self
+    }
+
+    pub fn manifest_directory(&mut self, directory: impl Into<PathBuf>) -> &mut Self {
+        self.input_params.manifest_directory = Some(directory.into());
+        self
+    }
+
+    pub fn trace(&mut self, trace: bool) -> &mut Self {
+        self.input_params.trace = trace;
+        self
+    }
+
+    pub fn log_level(&mut self, log_level: Level) -> &mut Self {
+        self.input_params.log_level = log_level;
+        self
+    }
+
+    pub fn no_schema(&mut self, no_schema: bool) -> &mut Self {
+        self.input_params.no_schema = no_schema;
+        self
+    }
+
+    pub fn coverage(&mut self, coverage: bool) -> &mut Self {
+        self.input_params.coverage = coverage;
+        self
+    }
+
+    pub fn force_local_target(&mut self, local_target: bool) -> &mut Self {
+        self.input_params.force_local_target = local_target;
+        self
+    }
+
+    pub fn optimize_with_wasm_opt(&mut self, options: &wasm_opt::OptimizationOptions) -> &mut Self {
+        self.input_params.wasm_optimization = Some(options.to_owned());
+        self
+    }
+
+    pub fn build(&mut self) -> Result<ScryptoCompiler, ScryptoCompilerError> {
+        ScryptoCompiler::from_input_params(&self.input_params)
+    }
+
+    // Returns output wasm file path
+    pub fn compile(&mut self) -> Result<PathBuf, ScryptoCompilerError> {
+        self.build()?.compile()
+    }
+
+    pub fn compile_workspace(&mut self) -> Result<Vec<PathBuf>, ScryptoCompilerError> {
```

Comment:
I've reimplemented `workspace` compilation, now it is based on `cargo`.
For item 2, I've implemented it that way it will be compiled only if member of a workspace has defined `[package.metadata.scrypto]` section in its `Cargo.toml` file. For packages specified with `package` input parameter this is not checked.